app/models/katello/ping.rb

```diff
@@ -129,8 +125,24 @@ def pulp_without_auth(url)
           fail _("Pulp message bus connection issue.")
         end
 
+        unless all_pulp_workers_present?(json)
+          fail _("Not all necessary pulp workers running.")
+        end
+
         json
       end
+
+      def all_pulp_workers_present?(json)
+        worker_ids = json["known_workers"].collect { |worker| worker["_id"] }
+        if worker_ids
```

Comment:
Actually, `collect` always returns a non-nil value. So this check will always evaluate to true.